x/qgb/keeper/keeper_attestation.go

```diff
@@ -44,10 +44,14 @@ func (k Keeper) StoreAttestation(ctx sdk.Context, at types.AttestationRequestI)
 // SetLatestAttestationNonce sets the latest attestation request nonce, since it's
 // expected that this value will only increase by one and it panics otherwise.
 func (k Keeper) SetLatestAttestationNonce(ctx sdk.Context, nonce uint64) {
-	if k.CheckLatestAttestationNonce(ctx) && k.GetLatestAttestationNonce(ctx)+1 != nonce {
-		panic("not incrementing latest attestation nonce correctly")
-	}
+	if k.CheckLatestAttestationNonce(ctx) {
```

Comment:
Sorry my proposal didn't include the negation but `CheckLatestAttestationNonce` returns `false` when the latest attestation nonce isn't set so we need to negate this check

```suggestion
	if !k.CheckLatestAttestationNonce(ctx) {
```